apple/swift-docc
# improve detection for inherited symbol references

Bug/issue #, if applicable: rdar://93203894

## Summary

Symbols for operator functions that include multiple dots (e.g. the
ClosedRange `...` operator) are currently curated into an unnamed
"Implementations" section due to improper string splitting behavior.
This enhances the detection two ways:

1. For symbols where the "source origin" is available in our given
   symbol graphs, it loads the name information from the parent symbol.
2. For symbols where the "source origin" is not available, it improved
   the string splitting behavior by dropping empty-string components
   before reading names from it.

This allows Comparable and Equatable implementations to curate their
inherited operators properly.

## Dependencies

None

## Testing

Steps:
1. Generate documentation for a module which includes a conformance to Comparable.
2. Ensure that the Comparable range operators are correctly organized into a "Comparable Implementations" page.
3. Repeat with a protocol that defines a custom ellipsis operator (e.g. the `....` operator in the `FancyProtocol` symbol graph added in this PR).

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ n/a ] Updated documentation if necessary
